### DISCUSSION TITLE
New Action Type: Present user list of possible actions

### DISCUSSION THREAD

**Discussion by iantrich** [home-assistant/frontend#7434] [Dashboard] [answered]
I've found myself having to remind other users quite often what tap/hold/double_tap actions do and even forgetting myself at times. For some common use cases I've used the `navigate` action to go to a new view dedicated to actions for that particular entity, which works great and is highly customizable but something baked into Lovelace could be great for most use cases.

So an example scenario is light where I have the actions to defined to
tap: toggle
hold: lower brightness
double_tap: raise brightness

With a list of possible actions as the tap or hold action we could then present the user with a list that would allow them to get a lot more finer control without having to build more complicated UIs or add additional buttons for single tasks. So for example for a light my hold action could then present a list of additional actions such as:

Turn On
Turn Off
1% Brightness
50% Brightness
100% Brightness
Turn off in 10 minutes

I don't know what would be the best way to present these, a dialog with a list of buttons perhaps?

So your final config would be something like:

```yaml
type: button
entity: light.bed
tap_action:
  action: toggle
hold_action:
  action: list
  actions:
    - name: Turn On
      action: call-service
      service: light.turn_on
      service_data:
        entity_id: light.bed
    - name: Turn Off
      action: call-service
      service: light.turn_off
      service_data:
        entity_id: light.bed
etc
```

**Comment by iantrich** [comment]
![Peek 2020-10-22 22-05](https://user-images.githubusercontent.com/1287159/96951740-c7938480-14b2-11eb-836a-a606eed964cf.gif)
I went ahead and coded this up while "watching" the presidential debate :)

**Comment by iantrich** [comment, marked as answer]
https://github.com/home-assistant/frontend/pull/7464

**Comment by balloob** [comment]
While you move on from long press and double tap to the next thing, we have to keep maintaining it indefinitely. So let's think about if this is the right solution and format. Maybe the solution is that certain cards are not meant to do multiple things etc.

**Reply by iantrich** [reply]
@balloob not sure I follow your last prompt? Are you suggesting that we limit which cards can use this new action?

**Reply by iantrich** [reply]
Wanted to circle back to this to get us both on the same page.

Looking back I don't think hold/double should have ever been added due to their lack of discoverability. That being said, I don't think that should prevent us from moving forward with better approaches which I consider this to be. At least to me, this is a clean and very easily maintained approach unlike the existing actions.

**Reply by balloob** [reply]
Let me rephrase: let's think through if this is going to solve the problem. I think that we were too hastily in adding long press and such features will live on forever.

I am not opposed to the idea. Styling can be improved.

**Reply by balloob** [reply]
The only thing we should consider is, how does this relate to the more info dialog. Can we go from here to more info? Or maybe we can incorporate quick actions into more info to begin with, so we don't need this?

**Reply by iantrich** [reply]
You can call the more-info from the list dialog (perhaps that could be a default button or something on the list dialog?).
Putting quick actions on the more info sounds interesting but we have some that are already pretty cluttered; light comes to mind with the color picker. adding quick actions to that would either push that content down or be really far down at the bottom. Could have a quick actions tab, perhaps?

**Reply by iantrich** [reply]
![image](https://user-images.githubusercontent.com/1287159/99717684-56be9880-2a6f-11eb-8585-b2741e2a692e.png)
That would be pretty slick. Would we default to details or actions? I would vote for defaulting to details but perhaps an option to start on actions?
```yaml
action: more-info
show_actions: true
actions:
  - name: Turn On
    action: call-service
    service: light.turn_on
    service_data:
      entity_id: light.bed
  - name: Turn Off
    action: call-service
    service: light.turn_off
    service_data:
      entity_id: light.bed
```

**Reply by BoneheadFraggle** [reply]
This would fit nicely with most of the structure from the ideas here :-) https://github.com/home-assistant/frontend/discussions/7515

**Comment by SeanPM5** [comment]
FWIW, I think this is a good idea that'd allow Home Assistant to reach a wider audience. 

It keeps the UI clean and simple because now you can replace six buttons with just one. And it makes the UX easier since now you don’t have to remember — or teach others — whether something was a tap, double tap, or hold action. You’d be getting the best of both worlds here, simplicity and flexibility.

This would make it a lot more approachable for children, seniors, and those unfamiliar with Home Assistant. A lot of those people would struggle with “secret” undiscoverable long press actions, but everyone knows how to press a button and select an action from a list. Eliminates any guesswork/confusion and makes things foolproof.

And I think this would really shine on wall mounted tablets and Home Assistant Cast.
